packages/oas-to-snippet/src/languages.ts

```diff
@@ -0,0 +1,367 @@
+      default: 'requests',
+      targets: {
+        requests: { name: 'Requests' },
+      },
+    },
+  },
+  r: {
+    highlight: 'r',
+    httpsnippet: {
+      lang: 'r',
+      default: 'httr',
+      targets: {
+        httr: { name: 'httr' },
+      },
+    },
+  },
+  ruby: {
+    highlight: 'ruby',
+    httpsnippet: {
+      lang: 'ruby',
+      default: 'native',
+      targets: {
+        native: { name: 'net::http' },
+      },
+    },
+  },
+  shell: {
+    highlight: 'shell',
+    httpsnippet: {
+      lang: 'shell',
+      default: 'curl',
+      targets: {
+        curl: {
+          name: 'cURL',
+          opts: {
+            escapeBrackets: true,
+            indent: '     ',
+          },
+        },
+        httpie: { name: 'HTTPie' },
+      },
+    },
+  },
+  swift: {
+    highlight: 'swift',
+    httpsnippet: {
+      lang: 'swift',
+      default: 'nsurlsession',
+      targets: {
+        nsurlsession: { name: 'NSURLSession' },
+      },
+    },
+  },
+};
+
+/**
+ * Retrieve all supported supported languages by this library. You can also optionally supply an
+ * array of `httpsnippet` plugins to extend this list.
+ *
+ */
+export function getSupportedLanguages(
+  {
+    plugins,
+  }: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    plugins?: ClientPlugin<Record<string, any>>[];
+  } = { plugins: [] },
+) {
+  const languages: SupportedLanguages = JSON.parse(JSON.stringify(DEFAULT_LANGUAGES));
+
+  Object.entries(targets).forEach(([target, { clientsById }]) => {
+    if (!(target in languages)) {
+      return;
+    }
+
+    Object.entries(clientsById).forEach(([client, data]) => {
+      if (!(client in languages[target as SupportedTargets].httpsnippet.targets)) {
+        return;
+      }
+
+      if ('installation' in data.info) {
+        languages[target as SupportedTargets].httpsnippet.targets[client].install = data.info.installation;
+      }
+    });
+  });
+
+  plugins.forEach(({ target, client: { info: clientInfo } }) => {
+    const clientKey = clientInfo.key;
+
+    languages[target as SupportedTargets].httpsnippet.targets[clientKey] = {
+      name: clientInfo.title,
+    };
+
+    if (clientInfo.installation) {
+      languages[target as SupportedTargets].httpsnippet.targets[clientKey].install = clientInfo.installation;
+    }
+  });
+
+  return languages;
+}
+
+export function getLanguageConfig(languages: SupportedLanguages, lang: Language) {
+  let config: LanguageConfig;
+  let language: TargetId;
+  let target: ClientId;
+
+  switch (lang) {
+    case 'curl':
+      config = languages.shell;
+      language = 'shell';
+      target = 'curl';
+      break;
+
+    // Our `api` snippet plugin, `httpsnippet-client-api`, that we use in ReadMe used to be called
+    // `node-simple`. For backwards compatibility we're retaining this old identifier here.
+    case 'node-simple':
+      config = languages.node;
+      language = 'node';
+      target = 'api';
+      break;
```

Comment:
noooo 😭 wait since this will be a breaking change anyways, can we remove this?

Reply:
there might still be specs in the wild with `node-simple` in the `x-samples-languages` extension.

Reply:
actually looks like we have code in the main app to remap `node-simple` to `[node, api]`. going to remove this then.

Reply:
also removing our backcompat for `curl` being rewritten to `shell` as we have that same logic upstream in our explorer